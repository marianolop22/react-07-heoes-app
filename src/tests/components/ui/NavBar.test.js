import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Navbar } from '../../../components/ui/NavBar';
import { AuthContext } from '../../../auth/AuthContext';
import { MemoryRouter, Router } from 'react-router-dom';
import { types } from "../../../types/types";

describe ('Pruebas de NavBar.test', ()=> {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }

    const contextValue = {
        user: {name:'Mariano', logged:true},
        dispatch: jest.fn()
    };

    const handleLogout = jest.fn();

    const wrapper = mount (
        <AuthContext.Provider value={
            contextValue
        }>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar/>
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach ( ()=> {
        jest.clearAllMocks();
    })

    test('debe de renderizar el objeto', () => {
        // console.log(wrapper.html());
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe(contextValue.user.name);
        //expect(contextValue.dispatch).toHaveBeenCalled();
    });

    test('debe de renderizar el objeto', () => {
        
        // wrapper.find('button').simulate('click'); 
        wrapper.find('button').prop('onClick')(); 

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type:types.logout
        });

        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    });

});