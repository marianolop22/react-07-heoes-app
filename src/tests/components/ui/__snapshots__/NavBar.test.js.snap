// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de NavBar.test debe de renderizar el objeto 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="navbar-text text-info"
      >
        Mariano
      </span>
      <button
        className="btn nav-item nav-link"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
